src/internal/NeoEsp8266AsyncUartMethod.h

```diff
@@ -0,0 +1,89 @@
+/*-------------------------------------------------------------------------
+NeoPixel library helper functions for using asynchronous UART1 in Esp8266.
+
+Written by Unai Uribarri
+
+I invest time and resources providing this open source code,
+please support me by dontating (see https://github.com/Makuna/NeoPixelBus)
+
+-------------------------------------------------------------------------
+This file is part of the Makuna/NeoPixelBus library.
+
+NeoPixelBus is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+NeoPixelBus is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with NeoPixel.  If not, see
+<http://www.gnu.org/licenses/>.
+-------------------------------------------------------------------------*/
+
+#pragma once
+
+#ifdef ARDUINO_ARCH_ESP8266
+#include "NeoEsp8266UartMethod.h"
+
+// due to linker overriding ICACHE_RAM_ATTR for cpp files, this function was moved into a NeoPixelEsp8266-AsyncUart.c file.
+extern "C" {
+    void ICACHE_RAM_ATTR esp8266_uart1_async_prepare();
+    void ICACHE_RAM_ATTR esp8266_uart1_async_send_pixels(uint8_t* pixels, uint8_t* end);
+}
+
+template<typename T_SPEED> class NeoEsp8266AsyncUartMethodBase: public NeoEsp8266UartMethodBase<T_SPEED>
+{
+public:
+    NeoEsp8266AsyncUartMethodBase(uint8_t pin, uint16_t pixelCount, size_t elementSize)
+	: NeoEsp8266UartMethodBase<T_SPEED>(pin, pixelCount, elementSize)
+    {
+        _asyncPixels = (uint8_t*)malloc(this->_sizePixels);
+    }
+
+    ~NeoEsp8266AsyncUartMethodBase()
+    {
+        while (!this->IsReadyToUpdate())
```

Comment:
Is this necessary in the destructor? Seems to me this doesn't _"do"_ anything besides waiting.